packages/frontend/src/components/chat/ChatContextMenu.tsx

```diff
@@ -0,0 +1,463 @@
+        0
+      )
+    }
+
+    // Prepare menu items
+
+    const { pin, archive } = buildArchiveAndPinMenuItems(
+      unselectChat,
+      accountId,
+      chatListItems,
+      tx,
+      selectedChatId
+    )
+
+    const muteMenuItem = buildMuteMenuItem(chatListItems, accountId, tx)
+
+    // View and Edit menu items
+    const viewEditMenuItems = buildViewEditMenuItems(
+      singleChat,
+      selectedChat,
+      openDialog,
+      openViewGroupDialog,
+      openViewProfileDialog,
+      accountId,
+      tx
+    )
+
+    const encryptionInfoItem = buildEncryptionInfoMenuItem(
+      singleChat,
+      tx,
+      openEncryptionInfoDialog
+    )
+
+    const ephemeralMessagesMenuItem = isMainView &&
+      selectedChat.canSend &&
+      selectedChat.chatType !== 'InBroadcast' &&
+      selectedChat.chatType !== 'Mailinglist' &&
+      selectedChat.isEncrypted && {
+        label: tx('ephemeral_messages'),
+        action: onDisappearingMessages,
+      }
+
+    const showBlockContactOption =
+      singleChat &&
+      !isGroup &&
+      !singleChat.isSelfTalk &&
+      !singleChat.isDeviceTalk &&
+      !isOutBroadcast &&
+      singleChat.chatType !== 'InBroadcast'
+
+    const showLeaveGroupOption =
+      singleChat &&
+      isGroup &&
+      singleChat.isEncrypted &&
+      singleChat.isSelfInGroup &&
+      !singleChat.isContactRequest
+
+    // Build the complete menu
+    const menu: (ContextMenuItem | false)[] = [
+      isMainView && {
+        label: tx('search_in_chat'),
+        action: onSearchInChat,
+      },
+      !isMainView && pin,
+      muteMenuItem,
+      archive,
+      { type: 'separator' },
+      ...(!isMainView ? viewEditMenuItems : []),
+      // Clone Group
+      singleChat &&
+        isGroup && {
+          label: tx('clone_chat'),
+          action: () => {
+            openDialog(CloneChat, {
+              setViewMode: 'createGroup',
+              chatTemplateId: singleChat.id,
+            })
+          },
+        },
+      // Leave channel
+      singleChat &&
+        singleChat.chatType === 'InBroadcast' &&
+        !singleChat.isContactRequest && {
+          label: tx('menu_leave_channel'),
+          action: () => onLeaveGroupOrChannel(singleChat),
+          danger: true,
+        },
+      showLeaveGroupOption && {
+        label: tx('menu_leave_group'),
+        action: () => onLeaveGroupOrChannel(singleChat),
+        danger: true,
+      },
+      ephemeralMessagesMenuItem,
+      // Block contact
+      showBlockContactOption && {
+        label: tx('menu_block_contact'),
+        action: () => onBlockContact(singleChat),
+        danger: true,
+      },
+      {
+        label: tx('menu_delete_chat'),
+        action: onDeleteChats,
+        danger: true,
+      },
+      isMainView && {
+        label: tx('clear_chat'),
+        action: onClearChat,
+        danger: true,
+      },
+      !isMainView && encryptionInfoItem,
```

Comment:
TBH this looks really odd after the "danger" buttons:

<img width="226" height="265" alt="Image" src="https://github.com/user-attachments/assets/4569d277-aba4-4a3e-bcfc-a37ae2322378" />